src/libs/actions/Session/index.js

```diff
@@ -711,7 +711,116 @@ function unlinkLogin(accountID, validateCode) {
         },
     }];
 
-    API.write('UnlinkLogin', {accountID, validateCode}, {optimisticData, successData, failureData});
+    API.write('UnlinkLogin', {
+        accountID,
+        validateCode,
+    }, {
+        optimisticData,
+        successData,
+        failureData,
+    });
+}
+
+function enableTwoFactorAuth() {
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: true,
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    API.write('EnableTwoFactorAuth', {enable: true}, {optimisticData, successData, failureData});
+}
+
+function disableTwoFactorAuth() {
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: true,
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    API.write('DisableTwoFactorAuth', {enable: false}, {optimisticData, successData, failureData});
+}
+
+function validateTwoFactorAuth(twoFactorAuthCode) {
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: true,
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
```

Comment:
It could be simplified by creating a base object without a value prop to reduce line of codes, then pass it with spread operator, something like:
```
    optimisticData:[...baseOnyxData, { value: true }]
```

Reply:
Thanks! Ended up with `toggleTwoFactorAuth`:

```js
function toggleTwoFactorAuth(enable) {
    const optimisticData = [ ... ];

    const successData = [ ... ];

    const failureData = [ ... ];

    API.write(enable ? 'EnableTwoFactorAuth' : 'DisableTwoFactorAuth', {}, {optimisticData, successData, failureData});
}
```